### PR TITLE
Fix empty init statements in Go control flow

### PR DESCRIPTION
**Pull request by nicoberling**
The current parser cannot parse control flow with empty init statements such as:

```Go
if ; true {
  doSomething()
}
```

because the `simpleStmt` rule includes an emtpy statement that consists of only a semicolon, so it expects two semicolons. This also leads it to successfully parse this incorrect for loop:

```Go
for ;; true ; {
  doSomething()
}
```

This pull request adds a new rule `terminatedSimpleStmt`, which includes the semicolon for  all productions of `simpleStmt` except for the empty statement, which already contains a semicolon. This seems to be the simplest way to fix this issue, but it does introduce a new rule.

### CODE REVIEW

File: golang/Go/GoParser.g4 [hunk 1]
```diff
@@ -147,21 +155,21 @@ fallthroughStmt: FALLTHROUGH;
 deferStmt: DEFER expression;
 
 ifStmt:
-	IF (simpleStmt SEMI)? expression block (
+	IF (terminatedSimpleStmt)? expression block (
```
Comment (KvanTTT) [inline golang/Go/GoParser.g4:158]: I suggest dropping useless parentheses:

```suggestion
	IF terminatedSimpleStmt? expression block (
```

File: golang/Go/GoParser.g4 [hunk 2]
```diff
@@ -147,21 +155,21 @@ fallthroughStmt: FALLTHROUGH;
...
 		ELSE (ifStmt | block)
 	)?;
 
 switchStmt: exprSwitchStmt | typeSwitchStmt;
 
 exprSwitchStmt:
-	SWITCH (simpleStmt SEMI)? expression? L_CURLY exprCaseClause* R_CURLY;
+	SWITCH (terminatedSimpleStmt)? expression? L_CURLY exprCaseClause* R_CURLY;
```
Comment (KvanTTT) [inline golang/Go/GoParser.g4:165]:
```suggestion
	SWITCH terminatedSimpleStmt? expression? L_CURLY exprCaseClause* R_CURLY;
```

File: golang/Go/GoParser.g4 [hunk 3]
```diff
@@ -147,21 +155,21 @@ fallthroughStmt: FALLTHROUGH;
...
 
 exprCaseClause: exprSwitchCase COLON statementList?;
 
 exprSwitchCase: CASE expressionList | DEFAULT;
 
 typeSwitchStmt:
-	SWITCH (simpleStmt SEMI)? typeSwitchGuard L_CURLY typeCaseClause* R_CURLY;
+	SWITCH (terminatedSimpleStmt)? typeSwitchGuard L_CURLY typeCaseClause* R_CURLY;
```
Comment (KvanTTT) [inline golang/Go/GoParser.g4:172]:
```suggestion
	SWITCH terminatedSimpleStmt? typeSwitchGuard L_CURLY typeCaseClause* R_CURLY;
```

File: golang/GoParser.g4 [hunk 4]
```diff
@@ -147,21 +155,21 @@ fallthroughStmt: FALLTHROUGH;
 deferStmt: DEFER expression;
 
 ifStmt:
-	IF (simpleStmt SEMI)? expression block (
+	IF (terminatedSimpleStmt)? expression block (
```
Comment (KvanTTT) [inline golang/GoParser.g4:158]:
```suggestion
	IF terminatedSimpleStmt? expression block (
```

File: golang/GoParser.g4 [hunk 5]
```diff
@@ -147,21 +155,21 @@ fallthroughStmt: FALLTHROUGH;
...
 		ELSE (ifStmt | block)
 	)?;
 
 switchStmt: exprSwitchStmt | typeSwitchStmt;
 
 exprSwitchStmt:
-	SWITCH (simpleStmt SEMI)? expression? L_CURLY exprCaseClause* R_CURLY;
+	SWITCH (terminatedSimpleStmt)? expression? L_CURLY exprCaseClause* R_CURLY;
```
Comment (KvanTTT) [inline golang/GoParser.g4:165]:
```suggestion
	SWITCH terminatedSimpleStmt? expression? L_CURLY exprCaseClause* R_CURLY;
```

File: golang/GoParser.g4 [hunk 6]
```diff
@@ -147,21 +155,21 @@ fallthroughStmt: FALLTHROUGH;
...
 
 exprCaseClause: exprSwitchCase COLON statementList?;
 
 exprSwitchCase: CASE expressionList | DEFAULT;
 
 typeSwitchStmt:
-	SWITCH (simpleStmt SEMI)? typeSwitchGuard L_CURLY typeCaseClause* R_CURLY;
+	SWITCH (terminatedSimpleStmt)? typeSwitchGuard L_CURLY typeCaseClause* R_CURLY;
```
Comment (KvanTTT) [inline golang/GoParser.g4:172]:
```suggestion
	SWITCH terminatedSimpleStmt? typeSwitchGuard L_CURLY typeCaseClause* R_CURLY;
```

**Review by KvanTTT — CHANGES_REQUESTED** [review]

File: golang/Go/GoParser.g4 (hunk 1 above)
Comment (nicoberling) [reply golang/Go/GoParser.g4]: Absolutely, thanks for noticing!

**Review by KvanTTT — APPROVED** [review]

**Pull request merged**